src/opnsense/scripts/openvpn/user_pass_verify.php

```diff
@@ -121,7 +120,15 @@ function do_auth($common_name, $serverid, $method, $auth_file)
     foreach (explode(',', $a_server['authmode']) as $authName) {
         $authenticator = $authFactory->get($authName);
         if ($authenticator) {
-            if ($authenticator->authenticate($username, $password)) {
+            $checkpassword = $password;
+            if ($pin!=false) { 
+                if (property_exists($authenticator,'passwordFirst') && $authenticator->passwordFirst) {
+                    $checkpassword = $pinpassword . $pin;
+                } else {
+                    $checkpassword = $pin . $pinpassword;
+                }
+            }
+            if ($authenticator->authenticate($username, $checkpassword)) {
```

Comment:
if we had a third argument for the pin this might be possible to handle gracefully inside the authenticator but I am not sure of the benefit of al of this CC @AdSchellevis

Reply:
I thought also about that, but it would impact more code. If needed I might just change the Auth backends to accept the pin.

Reply:
About #3290 that's exactly the point. Enabling the pin request does not handle the reverse option enabled.

Reply:
Should we maybe move the "SCRV1:" handling code inside the authenticator itself?

Reply:
easiest option is just to add 2 authenticators if you need both orders, use one for openvpn, the other for the ui. I don't think we need more glue to be honest, a note in the documentation about limitations should be more than enough.